src/algebra/ring/aut.lean

```diff
@@ -60,4 +61,26 @@ by refine_struct { to_fun := ring_equiv.to_mul_equiv }; intros; refl
 def to_perm : ring_aut R →* equiv.perm R :=
 by refine_struct { to_fun := ring_equiv.to_equiv }; intros; refl
 
+/-- The tautological action by the group of automorphism of a ring `R` on `R`.-/
+instance apply_mul_semiring_action {R : Type*} [ring R] : mul_semiring_action (ring_aut R) R :=
+{ smul := ($),
+  smul_zero := ring_equiv.map_zero,
+  smul_add := ring_equiv.map_add,
+  smul_one := ring_equiv.map_one,
+  smul_mul := ring_equiv.map_mul,
+  one_smul := λ _, rfl,
+  mul_smul := λ _ _ _, rfl }
+
+/--A multiplicative action of `ring_aut R` on `R`
+ and a group homomorphism `G →* ring_aut R` induce a multiplicative action of `G` on `R`.-/
+def mul_semiring_action.to_ring_aut {R : Type*} {G : Type*} [group G] [ring R] [f : G →*(ring_aut R)]  :
```

Comment:
This looks weird to me; based on the name, I'd expect something like:

https://github.com/leanprover-community/mathlib/blob/2a528d11813a4452202e79a3c2d82fd7db2bc6e8/src/algebra/algebra/basic.lean#L1344-L1355

Reply:
Oh, I didn't know about these. What should I do in this case?

Reply:
The name here is right, but it should be of type `G →* ring_aut R` and implemented as `{ to_fun := mul_semiring.to_ring_equiv _, map_one' := _ }` or whatever actually compiles

Reply:
Oh right, so this is the forward direction ? Thus, what I wrote is the reverse?

Reply:
I also need the fact that that ```mul_semiring_action G R``` is the same thing as ```G ->* ring_aut R```. 

So, what name could I give to that instance ?

Reply:
> The name here is right, but it should be of type `G →* ring_aut R` and implemented as `{ to_fun := mul_semiring.to_ring_equiv _, map_one' := _ }` or whatever actually compiles

Done. I will commit it tomorrow (Monday)

Reply:
Thank you. I just added  ```mul_semiring_action.to_ring_aut ```. Is there anything else I should do for this PR?